### PR TITLE
commons-compress 1.21 -> 1.24.0

### PR DESCRIPTION
**Pull request by mrbonha**
Test cases look good.
Release notes:
https://commons.apache.org/proper/commons-compress/changes-report.html#a1.22
https://commons.apache.org/proper/commons-compress/changes-report.html#a1.23.0
https://commons.apache.org/proper/commons-compress/changes-report.html#a1.24.0

Conflicting reports of availability.

Drafting pending further verification.

### CODE REVIEW

**Comment by mrbonha** [comment]
Tested through. All clear

File: pom.xml [hunk 1]
```diff
@@ -48,7 +48,7 @@
     <checkstyleFormatter>${project.basedir}/contrib/checkstyle.xml</checkstyleFormatter>
     <dep.commons-codec.version>1.15</dep.commons-codec.version>
     <dep.commons-collections.version>4.4</dep.commons-collections.version>
-    <dep.commons-compress.version>1.21</dep.commons-compress.version>
+    <dep.commons-compress.version>1.24.0</dep.commons-compress.version>
```
Comment (cfkoehler) [inline pom.xml:51]: Is it worth going to 1.25.0?

Comment (jpdahlke) [reply pom.xml:51]: Is that available?

Comment (cfkoehler) [reply pom.xml:51]: I don't know. I forgot to check. We can just go with 24 for now

**Review by jpdahlke — APPROVED** [review]

**Pull request merged**